Gruntfile.js

```diff
@@ -40,6 +40,36 @@ var getFromExec = function (command, options) {
     return result;
 };
 
+/**
+* Returns the "full extension" of a filename containing
+* multiple periods; common in minified/mapped JS file name
+* conventions, such as "infusion-all.min.js.map"
+* @param {String} filename - the filename to parse
+* @returns {String} the extracted extension such as ".min.js.map"
+*/
+var getFullFilenameExtension = function (filename) {
+    var firstPeriod = filename.indexOf(".");
+    return filename.slice(firstPeriod);
+};
+
+/**
+ * Rename function for grunt file tasks for  adding ".min" convention
+ * to filename string; won't do anything to strings that already
+ * include ".min"
+ * @param {String} dest - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+ * @param {String} src - supplied by Grunt task, see http://gruntjs.com/configuring-tasks#the-rename-property
+*/
+var addMinifyToFilename = function (dest, src) {
+    var fullExtension = getFullFilenameExtension(src);
+    var minifiedExtension = ".min" + fullExtension;
+    // Don't operate on files that already have a .min extension
+    if (fullExtension.indexOf(".min.") > -1) {
+        return dest + src;
+    } else {
+        return dest + src.replace(fullExtension, minifiedExtension);
+    }
+};
```

Comment:
What do you think about performing the string operation using arrays instead? See this jsfiddle example I quickly threw together https://jsfiddle.net/p7xdnbbz/.